app/src/main/java/com/T05/krowdtrialz/util/Database.java

```diff
@@ -339,6 +339,58 @@ public void onFailure(@NonNull Exception e) {
         });
     }// end getExperimentsBySearch
 
+    /**
+     * This method returns a list of experiments where given tags match description, owner name, owner username, or owner email, region
+     * and unit(s)
+     * @author
+     *  Furmaan Sekhon and Jacques Leong-Sit
+     * @param tags
+     *  This is the tags to search for
+     */
+    public void getExperimentsByTags (ArrayList<String> tags, QueryExperimentsCallback callback) {
```

Comment:
Assuming tags were changed from Set to ArrayList because firestore needs ArrayList? Could we convert the Set to ArrayList instead to avoid duplicates?

Reply:
I agree with Andrew's suggestion

Reply:
Yes, this is a good idea. Furmaan and I are meeting again at 8 so we will change it then.